dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiTitle, EuiComboBox, EuiSpacer, EuiButton, EuiFieldText, EuiFlexItem, EuiFormRow, EuiIcon, EuiPanel, EuiText } from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+import { render as renderExplorerVis } from '../../../../../../event_analytics/redux/slices/visualization_slice';
+import { selectExplorerVisualization } from '../../../../../../event_analytics/redux/slices/visualization_slice';
+import { AGGREGATION_OPTIONS } from '../../../../../../../../common/constants/explorer';
+import { ButtonGroupItem } from './config_button_group';
+import { visChartTypes } from '../../../../../../../../common/constants/shared';
+
+export const DataConfigPanelItem = ({
+  fieldOptionList,
+  visualizations,
+  tabID
+}: any) => {
+  const dispatch = useDispatch();
+  const explorerVisualizations = useSelector(selectExplorerVisualization)[tabID];
+
+  const { data } = visualizations;
+  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;
+
+  const newEntry = { label: "", aggregation: "", custom_label: "", name: "", side: "right" };
+
+  const [configList, setConfigList] = useState({
+    dimensions: [{ ...newEntry }],
+    metrics: [{ ...newEntry }],
+  });
+
+  useEffect(() => {
+    if (data.rawVizData?.dataConfig) {
+      setConfigList({
+        ...data.rawVizData?.dataConfig
+      })
+    } else if (data.defaultAxes.xaxis || data.defaultAxes.yaxis) {
+      const { xaxis, yaxis } = data.defaultAxes;
+      setConfigList({
+        dimensions: [...xaxis && xaxis],
+        metrics: [...yaxis && yaxis],
+      })
+    }
+  }, [data.defaultAxes, data.rawVizData?.dataConfig]);
+
+  const updateList = (value: string, index: number, name: string, field: string) => {
+    let list = { ...configList };
+    let listItem = { ...list[name][index] };
+    listItem = { ...listItem, [field]: value };
+    const newList = {
+      ...list,
+      [name]: [
+        ...list[name].slice(0, index),
+        listItem,
+        ...list[name].slice(index + 1, list[name].length),
+      ],
+    };
+    setConfigList(newList);
+  }
+
+  const onfieldOptionChange = (e, index: number, name: string) => {
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'label');
+   };
+
+  const onAggregationChange = (e, index: number, name: string) => {
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'aggregation');
+  };
+
+  const onCustomLabelChange = (e, index: number, name: string) => {
```

Comment:
can we combine all onfieldOptionChange , onAggregationChange, onCustomLabelChange in to one method as more of it is calling same method? Also , check if we really need this method as 
` onChange={(e) => updateList(label, index, name, sectionName);}` should work.